### PR TITLE
Move rhel-7 image to 7.4

### PR DESCRIPTION
**Pull request by martinpitt**
Now that 7.4 is released, "yum update"-ing the previous 7.3 image just
lifts it to 7.4, so start with 7.4 right away. Copy the necessary and
relevant changes from rhel-7-4.setup.
    
Move rhel-7-4 naughty issues to rhel-7 and make the former a symlink to
the latter.

yum install skips over nonexisting packages as long as at least one given package is valid. This produces broken images like in PR #7427. So double-check installed packages on RHEL.

 * [x] image-refresh rhel-7-4

### CODE REVIEW

**Comment by martinpitt** [comment]
The check-storage-resize failure was simple, it's just known issue #6960, I (locally) added the naughty override. The check-setroubleshoot one is more involved: It seems that you can now actually change the type of ~/.ssh/authorized_keys to `tmp_t` and SSH is still allowed to read it. I can run the `SELINUX_FIXABLE_ALERT_SCRIPT` steps manually, and ssh succeeds:
```
# chcon -t tmp_t ~/.ssh/authorized_keys
# ls -Z .ssh/authorized_keys
-rw-r--r--. root root unconfined_u:object_r:tmp_t:s0   .ssh/authorized_keys
# ssh -o StrictHostKeyChecking=no -o 'BatchMode=yes' -i ~/.ssh/test-avc-rsa localhost 'echo pwned'
pwned
```

I also tried `boot_t` and `cert_t`, same result.

**Comment by martinpitt** [comment]
One big difference is that `UsePrivilegeSeparation` in /etc/ssh/sshd_config is disabled (by default) on the old image and enabled on the new image. But flipping it around doesn't change behaviour on either image, so that's not it.

Versions have bumped quite a bit. Current image in master:
```
openssh-6.6.1p1-35.el7_3.x86_64
openssh-clients-6.6.1p1-35.el7_3.x86_64
openssh-server-6.6.1p1-35.el7_3.x86_64
selinux-policy-3.13.1-102.el7_3.16.noarch
selinux-policy-targeted-3.13.1-102.el7_3.16.noarch
```

This image:
```
# rpm -qa|egrep 'selinux-policy|openssh'|sort
openssh-7.4p1-11.el7.x86_64
openssh-clients-7.4p1-11.el7.x86_64
openssh-server-7.4p1-11.el7.x86_64
selinux-policy-3.13.1-166.el7.noarch
selinux-policy-targeted-3.13.1-166.el7.noarch
```

These jumps are mostly due to `openssh` and `selinux-policy` packages now coming from the `rhel-7-server-htb-rpms` repo instead of the `rhel-7-server-rpms` one. This new image has this "High Touch  Beta" product registered, which the current master image doesn't. I think we need to change this in our `subscription-manager` call.

**Comment by martinpitt** [comment]
Urgh, and even more importantly, on the rebuilt image the `Red Hat Enterprise Linux Server` product is now version 7.4, and so is `/etc/os-release`! I think the `yum update` lifted the original 7.3 installation to 7.4. Is there some way to prevent that and only install the 7.3 updates?

**Comment by martinpitt** [comment]
`yum update` updates a 7.3 installation to 7.4 and it's not a thing to "stay at 7.3" at this point - you just have "RHEL 7", not "7.x". After discussion with @dperpeet we'll keep the rhel-7-4 image with cockpit from base pre-installed, and will use the rhel-7 image for testing cockpit from master, but both will be RHEL 7.4. I'll rework this PR accordingly.

**Comment by martinpitt** [comment]
Reworked accordingly. New rhel-7 image still uploading.

File: bots/images/scripts/rhel-7-4.setup [hunk 1]
```diff
@@ -82,8 +82,6 @@ NetworkManager-team \
 PackageKit \
 pcp-libs \
 pcp \
-python-pip \
```
Comment (dperpeet) [inline bots/images/scripts/rhel-7-4.setup]: if the rhel-7-4 setup is changed, I think we should also build a new image and test that before merging

**Comment by cockpituous** [comment]
image-refresh in progress on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/image-refresh-7466-20170809-123636/

**Comment by cockpituous** [comment]
image-refresh rhel-7-4 done: https://github.com/cockpituous/cockpit/commits/image-refresh-rhel-7-4-20170809-130202

**Comment by martinpitt** [comment]
@dperpeet: rhel-7-4 image rebuilt, included, and test-triggered.

File: bots/images/scripts/rhel-7-4.setup [hunk 2]
```diff
@@ -115,12 +113,24 @@ valgrind \
 gdb \
 yum-utils \
 virt-install \
-qemu-system-x86 \
 qemu-kvm \
 "
 
 yum install --nogpgcheck -y $TEST_PACKAGES $COCKPIT_DEPS $IPA_CLIENT_PACKAGES
 
+# the above silently skips over unknown packages, so make sure we got everything
```
Comment (dperpeet) [inline bots/images/scripts/rhel-7-4.setup]: nice improvement!

File: bots/naughty/rhel-7-4 [hunk 3]
```diff
@@ -0,0 +1 @@
+rhel-7
```
Comment (dperpeet) [inline bots/naughty/rhel-7-4]: Is this logic the right way around? What happens when 7.5 is released? I would think that right now rhel-7 is rhel-7-4, but rhel-7-4 is more specific.

File: test/verify/check-setroubleshoot [hunk 4]
```diff
@@ -114,7 +114,7 @@ class TestSelinux(MachineCase):
 
         row_selector = "tbody:contains('sshd from open access on the file')"
 
-        if self.machine.image in ["rhel-7-4"]:
+        if self.machine.image in ["rhel-7", "rhel-7-4"]:
```
Comment (dperpeet) [inline test/verify/check-setroubleshoot]: as a followup I think we should get something in here that's fixable in 7.4

File: bots/naughty/rhel-7-4 (hunk 3 above)
Comment (martinpitt) [inline bots/naughty/rhel-7-4]: At that point we need to split it again; for now it seemed easier to keep the two in sync, but if you prefer I can also copy them right away.

File: test/verify/check-setroubleshoot (hunk 4 above)
Comment (martinpitt) [inline test/verify/check-setroubleshoot]: Yeah, agreed; "setroubleshoot too old" is either wrong or at least not telling the whole story - there is no violation with our test in the first place. I'm not SELinux savy enough to figure out how to get a violation, though.

File: bots/naughty/rhel-7-4 (hunk 3 above)
Comment (dperpeet) [inline bots/naughty/rhel-7-4]: It's ok for now, I thought you were moving the files over and then linking, but this is a fairly simple cleanup.

**Comment by dperpeet** [comment]
rhel-7 and rhel-7-4 have the same failures... are they flakes?

**Comment by martinpitt** [comment]
git pilot error, forgot to add the two remaining rhel-7 flakes. Fixed.

**Comment by martinpitt** [comment]
Green now!

**Pull request closed without merging**